src/libsyncengine/update_detection/file_system_observer/fsoperationset.cpp

```diff
@@ -84,52 +80,60 @@ void FSOperationSet::clear() {
 }
 
 void FSOperationSet::insertOp(FSOpPtr opPtr) {
-    const std::lock_guard<std::mutex> lock(_mutex);
+    const std::scoped_lock lock(_mutex);
     startUpdate();
-    _ops.insert({opPtr->id(), opPtr});
+    _ops.try_emplace(opPtr->id(), opPtr);
     _opsByType[opPtr->operationType()].insert(opPtr->id());
     _opsByNodeId[opPtr->nodeId()].insert(opPtr->id());
 }
 
 bool FSOperationSet::removeOp(UniqueId id) {
-    const std::lock_guard<std::mutex> lock(_mutex);
+    const std::scoped_lock lock(_mutex);
     startUpdate();
-    auto it = _ops.find(id);
-    if (it != _ops.end()) {
+    if (auto it = _ops.find(id); it != _ops.end()) {
         _opsByType[it->second->operationType()].erase(id);
+        if (_opsByType.find(it->second->operationType())->second.empty()) {
+            _opsByType.erase(it->second->operationType());
+        }
         _opsByNodeId[it->second->nodeId()].erase(id);
+        if (_opsByNodeId.find(it->second->nodeId())->second.empty()) {
+            _opsByNodeId.erase(it->second->nodeId());
+        }
         _ops.erase(id);
         return true;
     }
     return false;
 }
 
 bool FSOperationSet::removeOp(const NodeId &nodeId, const OperationType opType) {
-    FSOpPtr op = nullptr;
-    if (findOp(nodeId, opType, op)) {
+    if (FSOpPtr op = nullptr; findOp(nodeId, opType, op)) {
         return removeOp(op->id());
     }
     return false;
 }
 
-bool FSOperationSet::findOp(const NodeId &nodeId, const OperationType opType, FSOpPtr &res) {
-    const std::lock_guard<std::mutex> lock(_mutex);
-    auto it = _opsByNodeId.find(nodeId);
-    if (it != _opsByNodeId.end()) {
-        for (auto id : it->second) {
-            FSOpPtr opPtr = nullptr;
-            if (getOp(id, opPtr, false)) {
-                if (opPtr->operationType() == opType) {
-                    res = opPtr;
-                    return true;
-                }
-            }
+bool FSOperationSet::findOp(const NodeId &nodeId, const OperationType opType, FSOpPtr &res) const {
+    const std::scoped_lock lock(_mutex);
+    if (!_opsByNodeId.contains(nodeId)) {
+        return false;
+    }
+    for (auto it = _opsByNodeId.find(nodeId); auto id : it->second) {
+        FSOpPtr opPtr = nullptr;
+        if (getOp(id, opPtr) && opPtr->operationType() == opType) {
+            res = opPtr;
+            return true;
         }
     }
     return false;
 }
 
-FSOperationSet &FSOperationSet::operator=(const FSOperationSet &other) {
+ReplicaSide FSOperationSet::side() const {
+    const std::scoped_lock lock(_mutex);
```

Comment:
This lock is useless, variable `_side` is never changed

Reply:
[b3b1d7c](https://github.com/Infomaniak/desktop-kDrive/pull/203/commits/b3b1d7c8c92629981ee79b3407a15cc9e77051d7)